pkg/helm/helm_helpers.go

```diff
@@ -32,14 +33,21 @@ type k8sObj struct {
 type K8sObjectType string
 
 type Component struct {
-	k8sType K8sObjectType
-	name    string
+	k8sType      K8sObjectType
+	name         string
+	originalDump string // This holds the dry run string output of the resource
 }
 
+const (
+	K8sObjectTypeDeployment K8sObjectType = "deployment"
+)
+
+type FilterComponent func(name string, kind K8sObjectType) bool
+
 // ComponentsFromManifest is helper utility function that takes input the rendered output from dry-run enabled HelmApp Install and
 // return a slice of struct Component. This struct holds basic information about all the resources that are going to be deployed when
-// helm install is run in actual.
-func ComponentsFromManifest(manifest string) []Component {
+// helm install is run in actual. The second parameter adds criteria to filter out the components. Pass `nil` to fetch all.
+func ComponentsFromManifest(manifest string, filter FilterComponent) []Component {
```

Comment:
Let's move this function to the PR that was raised by Rajat. 
And in this PR we should just have the change that adds the helm test for taints, tolerations etc.

Reply:
Ok